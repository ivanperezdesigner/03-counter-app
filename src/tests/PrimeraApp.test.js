// const { render } = require("@testing-library/react")
const { default: PrimeraApp } = require("../PrimeraApp")
import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'

describe('Pruebas en <PrimeraApp /> ', () => {
    // test('Debe mostrar el mensaje "Hola"', () => {
    //     const saludo = 'Hola'
    //     const {getByText} = render(<PrimeraApp saludo={saludo} />)

    //     expect(getByText(saludo)).toBeInTheDocument()
    // })
    
    test('Debe mostrar el componente <PrimeraApp /> correctamente', () => {
        const saludo = 'Hola'
        const wraper = shallow(<PrimeraApp saludo={saludo}/>)
        expect(wraper).toMatchSnapshot()
    })
    
    test('Debe mostrar el subtítulo enviado por props', () => {
        const saludo = 'Hola'
        const subTitulo = 'Subtítulo de ejemplo'
        const wraper = shallow(<PrimeraApp saludo={saludo} subtitulo={subTitulo}/>)
        
        const textoParrafo = wraper.find('p').text()
        expect(textoParrafo).toBe(subTitulo)
    })
    
    
})
