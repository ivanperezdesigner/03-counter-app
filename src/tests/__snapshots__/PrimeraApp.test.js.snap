// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp />  Debe mostrar el componente <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola
  </h1>
  <p>
    Subt√≠tulo por defecto
  </p>
</Fragment>
`;
